docs/cpp/configure-intellisense.md

```diff
@@ -0,0 +1,123 @@
+---
+Order: 13
+Area: cpp
+TOCTitle: Configure IntelliSense
+ContentId: bf494c65-12b4-4506-ab6c-1fad76d7ccf1
+PageTitle: Configure C/C++ IntelliSense
+DateApproved: 8/23/2023
+MetaDescription: Configure Visual Studio Code IntelliSense in the C/C++ Extension
+
+---
+# Configure C/C++ IntelliSense
+
+This article is about configuring the C++ extension to provide C++ specific [IntelliSense](/docs/editor/intellisense.md) suggestions in VS Code. IntelliSense is a helpful tool built into VS Code that provides various code editing features that help you code faster and more efficiently. For example, code completion, parameter info, syntax highlighting, code actions (light bulbs), and member lists are all generated using IntelliSense.
+
+There are three types of sources that can be used for configuration:
+
+- A C/C++ compiler
+- `compile_commands.json` file
+- Another extension (for example, the [Makefile Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.makefile-tools) or [CMake extension](/docs/cpp/CMake-linux.md)).
+
+## When is IntelliSense automatically configured?
+
+The C++ Extension, in most occasions, configures IntelliSense automatically for you. For example, if you have previously selected a compiler for IntelliSense, that compiler will be set as your default compiler. Your default compiler is chosen to configure IntelliSense. Otherwise, if you have not previously selected a compiler, common compiler paths are queried by the C++ Extension. If a compiler is found in a world writeable location, that compiler is configured for IntelliSense. This applies to many common compilers in their default locations such as clang, GCC, MinGW, cygwin, cygwin64, and MSVC. If a compiler is not in a world writeable location, a notification is displayed asking to configure this compiler. When you approve a compiler for IntelliSense, double check that the compiler path stated is as expected.
```

Comment:
I don't think the people reading this document will know what "world writeable" means here. I'm confused myself, because I thought we did the opposite.  I thought the world writeable locations were the insecure ones.

The thing that might be most helpful to explain somewhere would be the reason why we need a compiler in the first place. The reason is to find the standard headers which define the STL.  When a compiler is picked to "configure" IntelliSense, we also search the entire workspace for headers by default.